sdk/core/azure-core/inc/azure/core/internal/diagnostics/log.hpp

```diff
@@ -139,16 +90,20 @@ namespace Azure { namespace Core { namespace Diagnostics { namespace _internal {
      * underlying stream object is flushed thus ensuring that the output is generated at the end of
      * the statement, even if the caller does not insert the std::endl object.
      */
-    class Stream {
+    class Stream final {
     public:
       /** @brief Construct a new Stream object with the configured I/O level.
        *
        * @param level - Represents the desired diagnostic level for the operation.
        */
-      Stream(Logger::Level level) : m_stream(GetStream(level)) {}
-
+      Stream(Logger::Level level) : m_level{level} {}
       /** @brief Called when the Stream object goes out of scope. */
-      ~Stream() { m_stream.flush(); }
+      ~Stream()
+      {
+        std::unique_lock<std::mutex> lock(m_mutex);
+        Log::Write(m_level, m_stream.str());
+        lock.unlock(); // Release the lock before destroying the mutex.
```

Comment:
Maybe a lock here is an overkill, but it is better to be safe.
`lock.unlock()` won't hurt either, however I don't understand why you need it here - after the unique_lock goes out of scope on the very next line, it will unlock.

Reply:
If I don't have the `lock.unlock()`, then the MSVC static analyzer complains that the lock is held when `m_mutex` is destroyed. It's really annoying, tbh.

If you'd prefer, I could just use scoping to achieve the same thing (add a `{` at line 103 and `}` at line 105).

Reply:
Thank you - both is good - either `{}`, or just update the comment to what you just said about the MSVC static analyzer, rather than the current comment.